cpp/ql/src/semmle/code/cpp/models/implementations/Send.qll

```diff
@@ -0,0 +1,55 @@
+/**
+ * Provides implementation classes modeling `send` and various similar
+ * functions. See `semmle.code.cpp.models.Models` for usage information.
+ */
+
+import semmle.code.cpp.models.interfaces.Taint
+import semmle.code.cpp.models.interfaces.ArrayFunction
+import semmle.code.cpp.models.interfaces.Alias
+import semmle.code.cpp.models.interfaces.FlowSource
+import semmle.code.cpp.models.interfaces.SideEffect
+
+/** The function `send` and its assorted variants */
+private class Send extends AliasFunction, ArrayFunction, SideEffectFunction, RemoteFlowFunctionSink {
+  Send() {
+    this.hasGlobalName([
+        "send", // send(socket, buf, len, flags)
+        "sendto", // sendto(socket, buf, len, flags, to, tolen)
+        "write" // write(socket, buf, len);
```

Comment:
There's also `sendmsg`, which is more complicated and doesn't necessarily belong in this class. I don't think we can fully model it with the vocabulary we have now, but we could at least say that its second argument is a sink.

Reply:
I didn't do it yet because we can't model `sendmsg`'s behavior correctly right now. But you're right in that we can definitely provide a version of a model that at least says that its second argument is a sink.